File: app/src/main/java/ru/nikshlykov/englishwordsapp/di/FragmentModule.java
package ru.nikshlykov.englishwordsapp.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ru.nikshlykov.englishwordsapp.ui.flowfragments.StudyFlowFragment;
import ru.nikshlykov.englishwordsapp.ui.fragments.GroupsFragment;
import ru.nikshlykov.englishwordsapp.ui.fragments.StatisticsFragment;
import ru.nikshlykov.englishwordsapp.ui.fragments.LinkOrDeleteWordDialogFragment;
import ru.nikshlykov.englishwordsapp.ui.fragments.SubgroupDataFragment;
import ru.nikshlykov.englishwordsapp.ui.fragments.SubgroupFragment;

@Module
public abstract class FragmentModule {

    @ContributesAndroidInjector
    abstract GroupsFragment contributeGroupsFragment();

    @ContributesAndroidInjector
    abstract StatisticsFragment contributeStatisticsFragment();

    @ContributesAndroidInjector
    abstract LinkOrDeleteWordDialogFragment contributeLinkOrDeleteWordDialogFragment();

    @ContributesAndroidInjector
    abstract StudyFlowFragment contributeStudyFlowFragment();

    @ContributesAndroidInjector
    abstract SubgroupDataFragment contributeSubgroupDataFragment();

    @ContributesAndroidInjector
    abstract SubgroupFragment contributeSubgroupFragment();
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/di/FragmentModule.kt
package ru.nikshlykov.englishwordsapp.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import ru.nikshlykov.englishwordsapp.ui.fragments.*

@Module
abstract class FragmentModule {
  @ContributesAndroidInjector
  abstract fun contributeGroupsFragment(): GroupsFragment

  @ContributesAndroidInjector
  abstract fun contributeStatisticsFragment(): StatisticsFragment

  @ContributesAndroidInjector
  abstract fun contributeLinkOrDeleteWordDialogFragment(): LinkOrDeleteWordDialogFragment

  @ContributesAndroidInjector
  abstract fun contributeSubgroupDataFragment(): SubgroupDataFragment

  @ContributesAndroidInjector
  abstract fun contributeSubgroupFragment(): SubgroupFragment

  @ContributesAndroidInjector
  abstract fun contributeWordFragment(): WordFragment

  @ContributesAndroidInjector
  abstract fun contributeAddWordFragment(): AddWordFragment

  @ContributesAndroidInjector
  abstract fun contributeProfileFragment(): ProfileFragment
}